### ISSUE TITLE
KeyboardAvoidingView stops working after rotation on iPad

### ISSUE THREAD

**Issue by mhoran** [facebook/react-native#47251]
### Description

Using the KeyboardAvoidingView on iPad with latest React Native 0.76.0 does not work after device rotation. The keyboard will add the desired padding (or height) prior to rotation when the keyboard is visible. However, when the device is rotated, the KeyboardAvoidingView stops working. This seems to only happen on iPad. It was not an issue in prior React Native versions.

### Steps to reproduce

1. Install the application **on an iPad simulator** with `yarn ios`
2. Select the text input and activate the software keyboard
3. Rotate the device
4. Note that the text input will no longer be visible and cannot be scrolled into view
5. Rotate the device back to the original orientation, text input still invisible
6. Text input will only become visible again after force quitting the app

### React Native Version

0.76.0

### Affected Platforms

Runtime - iOS

### Output of `npx react-native info`

```text
System:
  OS: macOS 14.7
  CPU: (16) arm64 Apple M3 Max
  Memory: 2.53 GB / 48.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.20.4
    path: /opt/homebrew/Cellar/node@18/18.20.4_1/bin/node
  Yarn:
    version: 1.22.22
    path: /opt/homebrew/bin/yarn
  npm:
    version: 10.7.0
    path: /opt/homebrew/Cellar/node@18/18.20.4_1/bin/npm
  Watchman: Not Found
Managers:
  CocoaPods:
    version: 1.15.2
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 24.0
      - iOS 18.0
      - macOS 15.0
      - tvOS 18.0
      - visionOS 2.0
      - watchOS 11.0
  Android SDK: Not Found
IDEs:
  Android Studio: Not Found
  Xcode:
    version: 16.0/16A242d
    path: /usr/bin/xcodebuild
Languages:
  Java: Not Found
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli":
    installed: 15.0.0-alpha.2
    wanted: 15.0.0-alpha.2
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.76.0
    wanted: 0.76.0
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: true
```

### Stacktrace or Logs

```text
None
```

### Reproducer

https://github.com/mhoran/keyboard-avoiding-view-repro/

### Screenshots and Videos

Prior to rotation:
![Simulator Screenshot - iPad Air 13-inch (M2) - 2024-10-28 at 12 17 44](https://github.com/user-attachments/assets/01e6b4ce-9afa-4d86-a1fb-b2e69e1db681)

After rotation:
![Simulator Screenshot - iPad Air 13-inch (M2) - 2024-10-28 at 12 18 35](https://github.com/user-attachments/assets/6120262f-2888-4842-b181-11cc1ed6075f)

**Comment by cipolleschi** [comment]
Was it working properly in 0.75?

**Comment by mhoran** [comment]
Reverting https://github.com/facebook/react-native/commit/3c54e1ee4522b26698bb3f99262a2a621b26fb64 resolves.

**Comment by mhoran** [comment]
It seems that `Dimensions` does not get updated reliably in 0.76, which is the root cause for this bug. I will open a separate issue and reproducer for that bug.

**Comment by mhoran** [comment]
Confirmed this is fixed by https://github.com/facebook/react-native/commit/e271b23fad89e1e7da2e91a8f1e5cbb295406c74 which is included in 0.76.1.
